mantle/kola/harness.go

```diff
@@ -533,6 +538,58 @@ func metadataFromTestBinary(executable string) (*externalTestMeta, error) {
 	return meta, nil
 }
 
+func runExternalTest(c cluster.TestCluster, mach platform.Machine) error {
+	var previousRebootState string
+	var stdout []byte
+	var stderr []byte
+	for {
+		bootID, err := platform.GetMachineBootId(mach)
+		if err != nil {
+			return errors.Wrapf(err, "getting boot id")
+		}
+		plog.Debug("Starting kolet run-test-unit")
+		if previousRebootState != "" {
+			plog.Debugf("Setting AUTOPKGTEST_REBOOT_MARK=%s", previousRebootState)
+			c.MustSSHf(mach, "echo AUTOPKGTEST_REBOOT_MARK=%s | sudo tee /run/kola-runext-env", previousRebootState)
+			previousRebootState = ""
+		}
+		// We ignore stderr here, it will end up in the journal
+		stdout, stderr, err = mach.SSH(fmt.Sprintf("sudo ./kolet run-test-unit %s", shellquote.Join(KoletExtTestUnit)))
+		if err != nil {
+			return errors.Wrapf(err, "kolet run-test-unit failed: %s", stderr)
+		}
+		koletRes := KoletResult{}
+		if len(stdout) > 0 {
+			err = json.Unmarshal(stdout, &koletRes)
+			if err != nil {
+				return errors.Wrapf(err, "parsing kolet json %s", string(stdout))
+			}
+		}
+		if koletRes.Reboot != "" {
+			previousRebootState = koletRes.Reboot
+			plog.Debugf("Reboot request with mark='%s'", previousRebootState)
+			// Stop sshd to ensure we don't race trying to log back in during shutdown
+			_, _, err = mach.SSH(fmt.Sprintf("sudo /bin/sh -c 'systemctl stop sshd && systemctl stop %s'", KoletRebootWaitUnit))
```

Comment:
I don't think we need to explicitly stop sshd here; the reboot code already knows how to handle this correctly. I.e. I think this can just be

```suggestion
			_, _, err = mach.SSH(fmt.Sprintf("sudo /bin/sh -c 'systemctl stop %s'", KoletRebootWaitUnit))
```

?

Reply:
You're right we don't need to; it seems better to do so, but eh.  Removed.

Reply:
OK I tried removing it and the force reboot seems to reliably trigger the 2 minute timeout failure.  I think what's happening in the force case is the client doesn't get a clean TCP connection close, so we'd need to wait for the TCP timeout.

Ultimately I think the `WaitForReboot()` API *should* handle this but...it's just way easier to stop sshd cleanly for now.